### PR TITLE
Implementing Thresholds for Incremental PR Reviews

### PR DESCRIPTION
**Pull request by zmeir**
## PR Type:
Enhancement

___
## PR Description:
This PR introduces new configurations to manage the frequency of automatic incremental reviews triggered by a push event. The main changes include:
- Added properties `first_new_commit_sha` and `last_seen_commit_sha` in the `IncrementalPR` class in `git_provider.py`.
- Modified `get_commit_range` method in `github_provider.py` to assign commit objects instead of commit SHA to `first_new_commit` and `last_seen_commit`.
- Updated `run` method in `pr_reviewer.py` to check for the number of new commits and their recency before starting the review.
- Added new configurations in `configuration.toml` to specify the minimum number of commits and minimum time elapsed for an incremental review.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

`pr_agent/git_providers/git_provider.py`: Introduced properties `first_new_commit_sha` and `last_seen_commit_sha` in the `IncrementalPR` class.
`pr_agent/git_providers/github_provider.py`: Updated `get_commit_range` method to assign commit objects instead of commit SHA to `first_new_commit` and `last_seen_commit`.
`pr_agent/tools/pr_reviewer.py`: Modified `run` method to check for the number of new commits and their recency before starting the review.
`pr_agent/settings/configuration.toml`: Added new configurations to specify the minimum number of commits and minimum time elapsed for an incremental review.
</details>

___
## User Description:
These new configurations allow limiting the rate of automatic incremental reviews when triggered by a push event to reduce the noise it may generate in the PR

### CODE REVIEW

**Comment by zmeir** [comment]
/describe --pr_description.add_original_user_description=true

**Review by okotek — APPROVED** [review]

File: pr_agent/tools/pr_reviewer.py [hunk 1]
```diff
@@ -100,9 +101,33 @@ async def run(self) -> None:
             if self.is_auto and not get_settings().pr_reviewer.automatic_review:
                 get_logger().info(f'Automatic review is disabled {self.pr_url}')
                 return None
-            if self.is_auto and self.incremental.is_incremental and not self.incremental.first_new_commit_sha:
-                get_logger().info(f"Incremental review is enabled for {self.pr_url} but there are no new commits")
-                return None
+            if self.incremental.is_incremental:
```
Comment (mrT23) [inline pr_agent/tools/pr_reviewer.py:104]: @zmeir  please move all of this to a dedicated, self-contained function.

this is too much content to appear directly

File: pr_agent/settings/configuration.toml [hunk 2]
```diff
@@ -25,6 +25,9 @@ inline_code_comments = false
 ask_and_reflect=false
 automatic_review=true
 remove_previous_review_comment=false
+require_all_thresholds_for_incremental_review=false
```
Comment (mrT23) [inline pr_agent/settings/configuration.toml:28]: can you stack all the 'incremental' parameters ?
something like

# incremental parameters:
minimal_commits_for_incremental_review=0
minimal_minutes_for_incremental_review=0
...

**Comment by mrT23** [comment]
@zmeir i gave some comments to fix.

in addition, this is a complicated ability.
Even after reviewing the code, I didn't fully understand what each parameter meant.
please add 3-4 lines of documentation the help make this ability accessible.

**Comment by zmeir** [comment]
@okotek @mrT23 Thanks for the review - I've implemented all of your suggestions, please let me know if anything else is missing, or if you have any further questions.

**Comment by hussam789** [comment]
@zmeir Thanks for adding this configuration / criteria!

**Pull request merged**